File: HR-simple-array-sum.js
//https://www.hackerrank.com/challenges/simple-array-sum/problem
//my submission: https://www.hackerrank.com/challenges/simple-array-sum/submissions/code/339201490
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function (inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function () {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}

/*
 * Complete the 'simpleArraySum' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts INTEGER_ARRAY ar as parameter.
 */

function simpleArraySum(ar) {
    //this seems like it could also be done recursively. Need to research, would prob be more efficient
    let counter = 0
    for (let i = 0; i < ar.length; i++) {
        console.log(counter + ar[i])
        counter = counter + ar[i]
    }
    return counter

}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const arCount = parseInt(readLine().trim(), 10);

    const ar = readLine().replace(/\s+$/g, '').split(' ').map(arTemp => parseInt(arTemp, 10));

    const result = simpleArraySum(ar);

    ws.write(result + '\n');

    ws.end();
}

File: HR-simple-array-sum.py
# https://www.hackerrank.com/challenges/simple-array-sum/problem
# WIP. Working on solving this recursively. 
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'simpleArraySum' function below.
#
# The function is expected to return an INTEGER.
# The function accepts INTEGER_ARRAY ar as parameter.
#

def simpleArraySum(ar):
    # Write your code here
    
    i = 0
    sum = 0
    #we make a function that takes in ar, i, and current sum
    def recursive_function(ar, i, sum):
        #base case is if  i >= ar.legnth, return the sum
        if i >= len(ar):
            return sum
        #else we add ar[i] to the current sum
        print(sum)
        sum += ar[i]
        print(sum)
        return recursive_function(ar, i+1, sum)
    
    #and we return function but with new params. 

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    ar_count = int(input().strip())

    ar = list(map(int, input().rstrip().split()))

    result = simpleArraySum(ar)

    fptr.write(str(result) + '\n')

    fptr.close()
